check_source.py

```diff
@@ -364,7 +343,33 @@ def source_has_correct_maintainers(self, source_project):
 
         return self.required_maintainer in maintainers
 
-    def select_add_role(self, request):
+    def __ensure_add_role_request(self, source_project):
+        """Declines request with link to add role request. Creates that add role if needed."""
+        declined_msg = (
+            'This request cannot be accepted unless %s is a maintainer of %s.' %
+            (self.required_maintainer, source_project)
+        )
+
+        try:
+            add_roles = get_request_list(self.apiurl, source_project,
+                req_state=['new', 'review'], req_type='add_role')
+            add_roles = list(filter(self.__select_required_maintainer, add_roles))
+            if len(add_roles) > 0:
+                add_role_reqid = add_roles[0].reqid
+            else:
+                add_role_msg = 'Created automatically from request %s' % self.request.reqid
+                add_role_reqid = create_add_role_request(self.apiurl, source_project, self.required_maintainer,
+                                                        'maintainer', message=add_role_msg)
+            declined_msg += ' Created the add_role request %s for addressing this problem.' % add_role_reqid
+        except HTTPError as e:
+            self.logger.error(
+                'Cannot create the corresponding add_role request for %s: %s' % (self.request.reqid, e)
+            )
+
+        self.review_messages['declined'] = declined_msg
+
+    def __select_required_maintainer(self, request):
+        """Selects add role requests that adds required maintainer user or group"""
```

Comment:
Nitpicking: this actually selects nothing (that's done by the filter function). This returns true if the action for the given request points to required_maintainer.